dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/TrackerPreheat.java

```diff
@@ -268,14 +269,15 @@ public MetadataIdentifier getCategoryOptionComboIdentifier( CategoryCombo catego
     private final Map<String, Relationship> relationships = new HashMap<>();
 
     /**
-     * Internal map of all relationships in the DB that are duplications of
-     * relationships present in the payload. Two relationship are duplicated if
-     * they have the same relationshipType, and the same relationshipItems, from
-     * and to. They key of the map is a string concatenating the
-     * relationshipType uid, the uid of the `from` entity and the uid of the
-     * `to` entity.
+     * Internal set of all relationship keys and inverted keys already present
+     * in the DB. This is used to validate only newly create relationships as
+     * update is not allowed for relationships. The key is a string
+     * concatenating the relationshipType uid, the uid of the `from` entity and
+     * the uid of the `to` entity. The inverted key is a string concatenating
+     * the relationshipType uid, the uid of the `to` entity and the uid of the
+     * `from` entity.
      */
-    private final Map<String, Relationship> duplicatedRelationships = new HashMap<>();
+    private final Set<String> duplicatedRelationships = new HashSet<>();
```

Comment:
I just thought that the name is a bit misleading. I mean we do not allow duplicate relationships. So `existingRelationships` is more accurate, no?